src/components/AddForm.js

```diff
@@ -12,35 +12,41 @@ function AddForm() {
   const [activitiesInput, setActivitiesInput] = useState("");
   const [locationsInput, setLocationsInput] = useState("");
   const [notesInput, setNotesInput] = useState("");
-  const [imageUpload, setImageUpload] = useState([]);
+  const [imageUploads, setImageUploads] = useState([]);
   const [imgPreview, setImgPreview] = useState([]);
   const history = useHistory();
 
   function handleSubmit(e) {
     e.preventDefault();
-    const formData = new FormData();
-    formData.append("file", imageUpload);
-    formData.append("upload_preset", "tyikvr8a");
+    const fileListAsArray = Array.from(imageUploads);
+    const imagesPromises = fileListAsArray.map((imageUpload) => {
+      const formData = new FormData();
 
-    fetch("https://api.cloudinary.com/v1_1/dyjecx1wm/image/upload", {
-      method: "PUT",
-      body: formData,
-    })
-      .then((response) => response.json())
-      .then((data) => {
-        const imageURL = data.secure_url;
+      formData.append("file", imageUpload);
+      formData.append("upload_preset", "tyikvr8a");
 
-        addTripsToLocalStorage({
-          id: destinationInput.split(" ").join("-"),
-          destination: destinationInput,
-          dates: datesInput,
-          activities: activitiesInput,
-          locations: locationsInput,
-          notes: notesInput,
-          photo: imageURL,
-        });
-        history.push("/trips");
+      return fetch("https://api.cloudinary.com/v1_1/dyjecx1wm/image/upload", {
+        method: "PUT",
+        body: formData,
+      }).then((response) => response.json());
+    });
+
+    Promise.all([...imagesPromises]).then(([...imagesResults]) => {
```

Comment:
You don't need to spread here since you are not combining anything in those arrays :)
```suggestion
    Promise.all(imagesPromises).then((imagesResults) => {
```